src/main/java/net/minecraftforge/common/extensions/IForgeItem.java

```diff
@@ -524,7 +525,8 @@ default void setDamage(ItemStack stack, int damage)
      */
     default boolean canPerformAction(ItemStack stack, ToolAction toolAction)
     {
-        return false;
+        // Temporary patch to keep #isShield working until its removed; change to `return false` once removed
+        return ToolActions.DEFAULT_SHIELD_ACTIONS.contains(toolAction) && isShield(stack, null);
```

Comment:
Minor, but I would invert the order. An instance comparison/boolean is faster than a hash set lookup.

This fallback backwards compat does feel a bit messy still, but I suppose not much to do about that.

Reply:
Done.